example/helloWorldLambda/src/helloWorldLambda.cpp

```diff
@@ -0,0 +1,213 @@
+    using Dim = alpaka::dim::DimInt<3>;
+    using Size = std::size_t;
+    using Host = alpaka::acc::AccCpuSerial<Dim, Size>;
+    using Acc = alpaka::acc::AccCpuSerial<Dim, Size>;
+    using Stream = alpaka::stream::StreamCpuSync;
+    using DevAcc = alpaka::dev::Dev<Acc>;
+    using DevHost = alpaka::dev::Dev<Host>;
+    using PltfHost = alpaka::pltf::Pltf<DevHost>;
+    using PltfAcc = alpaka::pltf::Pltf<DevAcc>;
+    using WorkDiv = alpaka::workdiv::WorkDivMembers<Dim, Size>;
+
+
+    /***************************************************************************
+     * Get the first devices
+     ***************************************************************************/
+    DevAcc const devAcc(alpaka::pltf::getDevByIdx<PltfAcc>(0u));
+    DevHost const devHost(alpaka::pltf::getDevByIdx<PltfHost>(0u));
+
+    
+    /***************************************************************************
+     * Create a stream to the accelerator device
+     ***************************************************************************/
+    Stream stream(devAcc);
+
+
+    /***************************************************************************
+     * Init workdiv
+     ***************************************************************************/
+    alpaka::vec::Vec<Dim, Size> const elementsPerThread(
+        static_cast<Size>(1),
+        static_cast<Size>(1),
+        static_cast<Size>(1));
+
+    alpaka::vec::Vec<Dim, Size> const threadsPerBlock(
+        static_cast<Size>(1),
+        static_cast<Size>(1),
+        static_cast<Size>(1));
+
+    alpaka::vec::Vec<Dim, Size> const blocksPerGrid(
+        static_cast<Size>(1),
+        static_cast<Size>(2),
+        static_cast<Size>(4));
+
+    WorkDiv const workdiv(
+        blocksPerGrid,
+        threadsPerBlock,
+        elementsPerThread);
+
+
+    /**
+     * Run "Hello World" kernel with lambda function
+     *
+     * Alpaka is able to execute lambda functions (anonymous functions) which
+     * are available since the C++11 standard.
+     * Alpaka forces the lambda function to accept
+     * the utilized accelerator as first argument. 
+     * All following arguments can be provided after
+     * the lambda function declaration or be captured. 
+     *
+     * This example passes the number exclamation marks, that should
+     * be written after we greet the world, to the 
+     * lambda function. Furthermore, the exclamation marks
+     * multiplier is captured directly by the lambda.
+     * 
+     * This kind of kernel function
+     * declaration might be useful when small kernels
+     * are written for testing or lambda functions
+     * allready exist.
+     */
+    const size_t nExclamationMarks = 10;
+    const size_t exclamationMarksMultiplier = 3;    
+    
+    auto const helloWorld(alpaka::exec::create<Acc>(
+        workdiv,
+	[&exclamationMarksMultiplier] ALPAKA_FN_ACC (Acc & acc, size_t const nExclamationMarks) -> void {
```

Comment:
I think that this is correct, as the reference to a host variable is not valid nor accessible on the device. You can only capture by copy (if it is allowed at all).